### PR TITLE
Setup a fallback for Select2 not having a locale

### PR DESCRIPTION
**Pull request by MatthewKennedy**
Avoid hitting a missing local if Select2 doesn't have the exact translation that matches the locale, it now falls back to en.

- Select 2 Translations
- Flatpickr Translations

### Flatpickr Fix
Move `date` or `date_time`  formatting onto each instance, rather than the javascript that initiates the instance, allows the developer to set a `date` or `date_time` selector as used in the Promotion form.

### Flatpickr Quirk
Respected Taiwanese as it was well translated and it seems a shame to not use it in Flatpicker, unfortunately it required a little extra code as the file path doesn't math the local naming convention.

### CODE REVIEW

File: backend/app/helpers/spree/admin/base_helper.rb [hunk 1]
```diff
@@ -201,6 +201,23 @@ def taxon_preview_link(taxon)
           class: 'btn-outline-secondary', icon: 'view.svg', id: 'admin_preview_taxon', target: :blank
         )
       end
+
+      def select2_local_fallback
+        stripped_locale = I18n.locale.to_s.split('-').first
+        all_select2_locales = %w[
```
Comment (damianlegawiec) [inline backend/app/helpers/spree/admin/base_helper.rb:207]: @MatthewKennedy can we move this into `SELECT2_SUPPORTED_LOCALES` static and freeze it?

**Comment by damianlegawiec** [comment]
@MatthewKennedy it would be awesome to do the same for FlatPickr - thank you!

**Comment by MatthewKennedy** [comment]
Will sort this today. Just running some errands first.

**Comment by damianlegawiec** [comment]
Screenshot 2021-02-04 at 11 30 53 AM
Screenshot 2021-02-04 at 11 30 32 AM
Screenshot 2021-02-04 at 11 30 02 AM

I've stumbled into those issues when using german as the admin panel language via

```
Spree::Backend::Config[:locale] = :de
```
in `config/initializers/spree.rb`

**Comment by MatthewKennedy** [comment]
@damianlegawiec are the translations coming from the Spree_i18n gem, I created a PR for that gem to take care for the date formatting issues.

https://github.com/spree-contrib/spree_i18n/pull/868

**Comment by damianlegawiec** [comment]
@MatthewKennedy got it

1. Could you rebase your branch with the master branch? I'll merge the I18n PR in a sec
2. Select2 dropdowns placeholders do not translate though

**Comment by MatthewKennedy** [comment]
@damianlegawiec The translation for choose_an_option needs adding to the files in i18_n

**Comment by MatthewKennedy** [comment]
I'll do that today.

**Comment by MatthewKennedy** [comment]
@damianlegawiec All should be good now.

**Comment by damianlegawiec** [comment]
Thank you 🙏

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**